test/gtest/ucp/ucp_test.cc

```diff
@@ -793,6 +828,73 @@ void ucp_test_base::entity::ep_destructor(ucp_ep_h ep, entity *e)
     ucp_request_release(req);
 }
 
+template <typename T>
+static void set_server_ep_param(T &param, bool is_set, const T &new_param,
+                                bool override, const char *param_name) {
+
+    if (!is_set || override) {
+        if (is_set) {
+            UCS_TEST_MESSAGE << "overriding " << param_name << " for server ep";
+        }
+
+        param = new_param;
+    } else {
+        UCS_TEST_MESSAGE << "ignoring " << param_name << " for server ep";
+    }
+}
+
+void ucp_test_base::entity::add_server_ep_params(const ucp_ep_params_t &params,
```

Comment:
instead of adding this function, test_ucp_sockaddr() should control the input params which are passed to listen() function.
for example, define 'virtual get_server_ep_params()' which would be overridden in test_ucp_sockaddr_destroy_ep_on_err, and no need to merge parameters.